src/shogun/distance/Distance.cpp

```diff
@@ -341,93 +266,69 @@ SGMatrix<T> CDistance::get_distance_matrix()
 	int32_t n=get_num_vec_rhs();
 
 	int64_t total_num = int64_t(m)*n;
+	int64_t total=0;
+	int64_t total_start=0;
+	int64_t total_end=total_num;
 
 	// if lhs == rhs and sizes match assume k(i,j)=k(j,i)
 	bool symmetric= (lhs && lhs==rhs && m==n);
 
 	SG_DEBUG("returning distance matrix of size %dx%d\n", m, n)
 
-		result=SG_MALLOC(T, total_num);
+	result=SG_MALLOC(T, total_num);
 
-	// TODO: port this to use OpenMP
-#ifdef HAVE_PTHREAD
-	int32_t num_threads=parallel->get_num_threads();
+	int32_t num_threads;
+	int64_t step;
+	#pragma omp parallel shared(num_threads, step)
+	{
+#ifdef HAVE_OPENMP
+		#pragma opm single
+		{
+			num_threads=omp_get_num_threads();
```

Comment:
@karlnapf no because if you check Parallel it is active when either openmp or threading backend is available ;) and we want to activate this when openmp is available.